### ISSUE TITLE
Add support for rapidsai - cudf and cuml packages GPU Data Science

### ISSUE THREAD

**Issue by jperez999** [Kaggle/docker-python#594]

**Comment by jperez999** [comment]
I have a PR ready for review with passing test cases for both cuml and cudf.

**Comment by rosbo** [comment]
@jperez999 

There is a few conflicts that prevent us from adding `cuDF` and `cuML`:

`cudf=0.10 -> pandas[version='>=0.24.2,<0.25']`

We are using pandas 0.25.3 and we have several packages requiring >= 0.25. Any particular reasons for requiring this restriction?

`cudf=0.10 -> pyarrow=0.14.1`

We are using pyarrow 0.15.1 and we also have several packages requiring >= 0.15. Any particular reasons for not allowing pyarrow 0.15.x?



```

**Comment by jperez999** [comment]
The reason we are running pandas < 0.25 https://github.com/rapidsai/cudf/pull/3486 And pyarrow lockdown is because of a conda-forge conflict https://github.com/rapidsai/cudf/pull/3318

**Comment by rosbo** [comment]
Hi @jperez999 and @kkraus14 

Are you planning on relaxing the constraint on `pyarrow` added in https://github.com/rapidsai/cudf/pull/3318? This is causing a cascading slew of downgrades the `arrow-cpp` lib and `boost-cpp` and so on.

Our image is currently using `pyarrow` 1.16 and `libboost` 1.72.0.

Thanks

**Comment by kkraus14** [comment]
We are planning on upgrading to Arrow 0.17.1 in cudf 0.15 shortly. Unfortunately there was a bug introduced in 0.15.1 of Arrow that made it incompatible with nvcc which made us unable to upgrade until now.

**Comment by kkraus14** [comment]
Hi @rosbo, we've upgraded to 0.17.1 in our nightlies but Arrow 1.0.0 has since released that we're considering upgrading to. Would that be a blocker for the kaggle container?

**Comment by rosbo** [comment]
Hi @kkraus14,

It shouldn't be a blocker, arrow 0.17.1 and 1.0.0 both work with our version of the boost library (cause of the cascading changes earlier).

Thanks for checking.

**Comment by kkraus14** [comment]
@rosbo hey rosbo, any chance we could give this another shot? The 0.15 release uses Arrow 0.17.1 and boost 1.72 and has been released for a bit now.

**Comment by rosbo** [comment]
Kicked off a new build with 0.15 and will see if we get any conflicts: https://github.com/Kaggle/docker-python/tree/add-rapids-ai-0.15

**Comment by rosbo** [comment]
We are in the process of migrating our GPU image to be based on the gcr.io/deeplearning-platform-release/tf2-gpu.2-3. 

Hitting conflicts again and I am waiting on conda to give me the list of conflicts... Will report hopefully soon.

```
docker run -it --rm gcr.io/deeplearning-platform-release/tf2-gpu.2-3:latest
$ conda install -c rapidsai -c nvidia -c conda-forge -c defaults rapids=0.15 python=3.7 cudatoolkit=10.1
```

**Comment by rosbo** [comment]
The conflict resolver did eventually converge after more than 72h and printed a long list of conflict (longer than my bash terminal history limit setting). I will take a look at this more closely once  we have migrated to the new base image.

**Comment by kkraus14** [comment]
Thanks @rosbo. We're planning on releasing 0.16 in ~2 weeks and it will upgrade Arrow to 1.0.1 and keep boost at 1.72.0 as the main version pinnings. Let me know if there's anything I can help with.

**Comment by kkraus14** [comment]
@rosbo quick ping here. 0.16 has been out for a while, any chance we could give this another shot?

**Comment by rosbo** [comment]
Hi @kkraus14, I will try to give it another shot this week. Thanks for the ping about the new release.

**Comment by rosbo** [comment]
@kkraus14 Great news!

I was able to successfully install `cudf` 0.16 on our images (no conflicts). I tried with only `cudf` first to reduce # of conflicts. I will try next to with `cuml`.

The original request was about adding `cudf` & `cuml`. I see now that Rapids includes several others packages. Should I try installing all of them with `conda install rapids` or just installed `cudf` and `cuml`?

Thank you
